### ISSUE TITLE
Bedrock edition support??

### ISSUE THREAD

**Issue by kristibektashi** [alexsocha/luckyblock#29]
The website mentions support for Bedrock Edition is "coming soon". When will it be ready?

**Comment by alexsocha** [comment]
Good question! Most of the planning is done - it's also part of the reason why the new (Fabric) version was rewritten in Kotlin and split into separate modules. The implementation is tricky though, since the bedrock scripting API is much more restrictive than Forge/Fabric.

I won't be able to implement everything, but I think I can get pretty close, including support for add-ons (add-on-add-ons?). I'll start development in a month, and, if you really want an estimate, I'll finish 3 months after that.

**Comment by kristibektashi** [comment]
It's been a month @alexsocha

**Comment by kristibektashi** [comment]
@alexsocha Well it's July now. How is the Bedrock Edition support coming along?

**Comment by alexsocha** [comment]
There were some delays in getting started, but it's being worked on now. We'll see how long it takes, but I wouldn't expect anything soon

**Comment by kristibektashi** [comment]
@alexsocha Well it's August now. Any updates?

**Comment by alexsocha** [comment]
Yes I'm making progress, but very slowly. I'd estimate that it's 50% done. Here's a screenshot of a lucky pig in bedrock edition:
<img width="1105" alt="lucky-pig" src="https://user-images.githubusercontent.com/5567163/129437802-0eedf7fa-626f-4d40-9422-4400f2b23673.png">

**Comment by kristibektashi** [comment]
Ok thanks I'll be back in September

**Comment by teddyxlandlee** [comment]
Well it _is_ September @kristibektashi

**Comment by kristibektashi** [comment]
> Well it _is_ September @kristibektashi

Oh right sorry I forgot

@alexsocha How is Bedrock Edition Support going?

**Comment by alexsocha** [comment]
I've added a progress section to my original comment, I'll update that at least once a month

**Comment by kristibektashi** [comment]
@alexsocha Any progress?

**Comment by Brandonbr1** [comment]
@alexsocha can we get an update on progress pls

**Comment by alexsocha** [comment]
Minecraft has removed the JS modding API: https://www.minecraft.net/en-us/creator/article/removing-the-additional-modding-capabilities-feature. I still think the project is possible, but many parts need to be reworked

**Comment by kristibektashi** [comment]
@alexsocha Any update?

**Comment by kristibektashi** [comment]
@alexsocha Since today I randomly remembered this mod exists, I am once again asking for your progress in Bedrock Edition support for Lucky Block mod
